### PR TITLE
fix: hashtag alignment fix

### LINKED ISSUE #198: Audit the RTL layout with the Arabic articles
@islamh-mahfouz, I was wondering if you could kindly audit the RTL layout with the Arabic articles and create an umbrella issue so we could address all of the inconsistencies together.

Here is what we have came across so far:
- The word break issue explained above.
- Twitter button tweets left to right titles.

_Originally posted by @ahmadabdolsaheb in https://github.com/freeCodeCamp/news/issues/194#issuecomment-1044378083_

### PR DESCRIPTION
**Pull request by zairahira**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Addresses #198

The hashtags above post titles are not aligned properly as the hashtag is coming at the end.

### CODE REVIEW

**Comment by scissorsneedfoodtoo** [comment]
Hey @zairahira, thanks for your patience.

I tested this locally, and unfortunately it doesn't seem to solve the issue we discussed in our last meeting.

All of the tags for the four views that show them (landing page cards, search results page cards, above post titles, tag page titles and popular tags) have the same issue, and show tags with the hashtag after the tag name and other text:

image

image

I believe it's because the CSS `direction` property really only works on block level elements, and inline elements like anchors need to have the `dir` attribute set directly.

The more robust fix would be to use the `dir` attribute in HTML elements as described in [this article](https://www.w3.org/International/questions/qa-html-dir).

But I think we can set some attributes directly now since tag names for RTL publications will always be in English like you, @Sboonny, and I talked about.

Here are some changes we can make in this PR rather than updating `screen.css`.

- `card.njk` (used for cards on the landing, tag, and author pages):

```
{% if primaryTag %}
    <span class="post-card-tags">
        <a dir="ltr" href="{{ primaryTag.path | url }}">
            #{{ primaryTag.name }}
        </a>
    </span>
{% endif %}
```

- `tag.njk`:

```
<h1 dir="ltr" data-test-label="tag-name" class="site-title tags-title">#{{ tag.name | upper }}</h1>
```

for the tag name heading, and

```
<a class='tag' href="{{ tag.path | url }}">
    <h3 dir="ltr">#{{ tag.name }} <small> | {{ tag.count.posts }}</small></h3>
</a>
```

for the list of popular tags.

- `post.njk` (the primary tag name above the post title):

```
<a dir="ltr" href="{{ primaryTag.path | url }}">
    #{{ primaryTag.name }}
</a>
```

- `search-results.js`:

```
${
  hit?.tags[0]?.name
    ? `<span class="post-card-tags"><a dir="ltr" href="${hit.tags[0].url}">
    #${hit.tags[0].name}
    </a></span>`
  : ''
}
```

All of these updates seem to fix tags in all the views that show them.

Would you like to make these updates and test them with a local build? Or would it be easier if I push some commits to your fork?

**Comment by zairahira** [comment]
Kris, thank you for sharing the feedback. The changes for `screen.css` have been reverted for now.

I have tested the changes locally and they are looking good.

image



image

File: src/search-results.njk [hunk 1]
```diff
@@ -51,3 +51,4 @@ permalink: '/search/'
     <meta property="og:image:width" content="{{ site.image_dimensions.cover_image.width }}">
     <meta property="og:image:height" content="{{ site.image_dimensions.cover_image.height }}">
 {% endblock %}
+
```
Comment (scissorsneedfoodtoo) [inline src/search-results.njk]: Noticed an extra newline at the end of this file:

```suggestion
```

**Review by scissorsneedfoodtoo — CHANGES_REQUESTED** [review]
Hey @zairahira, thanks for making all of these updates and for those helpful screenshots.

I pulled your latest changes down and can confirm that the tag names are fixed in most places.

The one place it's still going the wrong direction is on the search results page:

image

The fix would be to set `dir` to `ltr` in the anchor on line 86 of `search-results.js`.

Here's what that section of code looks like:

```js
${
  hit?.tags[0]?.name
    ? `<span class="post-card-tags"><a dir="ltr" href="${hit.tags[0].url}">
    #${hit.tags[0].name}
    </a></span>`
  : ''
}
```

Could you go back in and do that when you get a chance?

**Comment by zairahira** [comment]
@scissorsneedfoodtoo my bad, I forgot to push those changes. I have done them now.

**Review by scissorsneedfoodtoo — APPROVED** [review]
Tested locally and everything LGTM 👍 

Thanks again for reporting this, @zairahira, and for all your help fixing these alignment issues.

**Pull request merged**